collections/soldiers.js

```diff
@@ -0,0 +1,29 @@
+const MongoClient = require('mongodb').MongoClient;
+const assert = require('assert');
+
+// Connection URL
+const url = 'mongodb://localhost:27017';
+
+// Database Name
+const dbName = 'myproject';
+
+// Use connect method to connect to the server
+function newSoldier(data, callback) {
+  if (!(/[0-9]{7}/.test(data["id"]) && data["name"] && data["rank"] && data["limitations"])) {
+    callback("The soldier json that was sent is ileagal", null);
```

Comment:
i would prefer a more meaningfull message. telling the user what the problem was